### DISCUSSION TITLE
Hw 3 Part 1 Number 1

### DISCUSSION THREAD

**Discussion by ecieszynski** [cooperrc/computational-mechanics#51] [Q&A]
I imported t, x, and y from the reading and tried doing the polyfit, but t and vx have different lengths. Don't know why this happened because this wasn't a problem during the reading.

![image](https://user-images.githubusercontent.com/106542680/173480195-bdf2ca93-7018-4d32-886f-b4a9143eaece.png)

**Comment by cooperrc** [comment]
When you take a derivative, you lose `1` index e.g. 

`x = [1, 2, 5, 9]`
`dx = [1, 3, 4]`

use `t[1:]` or `t[0:-1]`

**Reply by ecieszynski** [reply]
okay, I solved that issue now, thanks. Got a new error though when using polyfit

![image](https://user-images.githubusercontent.com/106542680/173844819-a0331251-58f3-4cd9-87d1-5b38eb58b204.png)

**Reply by Aidan-Myatt** [reply]
For polyfit, you need to assign two variable to it. Like: Vx1, Vx0 = np.polyfit(dt, v_x, 1). Then, those two variable are used to assign it to a function like Vx_1d = np.poly1d((Vx1, Vx0)).

**Reply by cooperrc** [reply]
the warning is coming up because `dt` is all the same values right now, e.g. `[0.1, 0.1, 0.1, ..., 0.1]`

Use `t[1:] as your x-values.

**Reply by cooperrc** [reply]
the data is collected as 
|time (s)| x-position (m)|y-position (m)
|---|---|---|
|...|...|...|

We're looking for acceleration so we can compare
1. fit the x- and y-data to a parabola e.g. $x(t) = x_0 + v_{x0}t - \frac{a_x}{2}t^2$ <- find $a_x$
2. take a numerical derivative (`(x[1:] - x[:-1])/(t[1:] - t[:-1])`) then fit $v_x(t) = v_{x0} - a_x t$

**Reply by ecieszynski** [reply]
I think I got it now. Is this what I'm looking for?

![image](https://user-images.githubusercontent.com/106542680/174628880-5caabe00-d999-4521-8933-4bf7f01b4a09.png)
![image](https://user-images.githubusercontent.com/106542680/174628928-eb9eae73-36a4-4b4e-8347-b2a46c5a742a.png)

**Reply by cooperrc** [reply]
Perfecto :+1: :ok_hand:

**Reply by ecieszynski** [reply]
perfect. Also, I sent you a pretty lengthy email friday with some other questions, if you could check that out when you get a chance
